analysis_driver/util/__init__.py

```diff
@@ -45,3 +49,60 @@ def _write_bcbio_csv(run_dir, sample_id, fastqs, user_sample_id):
             writer.writerow([fq, user_sample_id])
 
     return csv_file
+
+
+def check_bcl_files(run_dir, run_info, validation_log):
+    tile_ids = run_info.tiles
+    run_info_ncycles = sum(int(e.attrib['NumCycles']) for e in run_info.mask.reads)
+    all_bcls = []
+
+    validated_bcls = []
+    if isfile(validation_log):
+        with open(validation_log, 'r') as f:
+            validated_bcls = [l.rstrip('\n') for l in f.readlines()]
+
+    all_cycles = _all_cycles(run_dir)
+    uniq_cycles = sorted(set(all_cycles), reverse=True)
+    if all_cycles[-1] > run_info_ncycles:
+        raise AnalysisDriverError(
+            'Number of cycles (%s) disagrees with RunInfo (%s)' % (all_cycles[-1], run_info_ncycles)
+        )
+
+    last_completed_cycle = None
+    for c in uniq_cycles:
+        if all_cycles.count(c) == len(tile_ids):
+            last_completed_cycle = c
+            break
+    if last_completed_cycle is None:
+        raise AnalysisDriverError('Bad cycles')
```

Comment:
if `last_completed_cycle` is `None` then it means that the first cycle is not finished isn't it ?
We risk of failing the process because we're too early in the process
maybe set `last_completed_cycle = 0`